### ISSUE TITLE
[Bug] Missing Hash key url in /api/v1/comments when reading comment threads

### ISSUE THREAD

**Issue by D3V1LC0D3R** [iv-org/invidious#2262]
**Describe the bug**
comments (comment threads not the original comments) sometimes fail to load with strange error
 
**Steps to Reproduce**
1. Go to any video
2. Click on load many comment threads
3. See error

**Logs**
the API response:
`{"error":"Missing hash key: \"url\"","errorBacktrace":"Missing hash key: \"url\" (KeyError)\n  from /usr/share/crystal/src/hash.cr:1030:9 in '[]'\n  from src/invidious/comments.cr:576:71 in 'content_to_comment_html'\n  from src/invidious/comments.cr:539:43 in 'parse_content'\n  from src/invidious/comments.cr:149:69 in 'fetch_youtube_comments'\n  from src/invidious.cr:2079:18 in '->'\n  from lib/kemal/src/kemal/route.cr:13:9 in '->'\n  from /usr/share/crystal/src/primitives.cr:255:3 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from src/invidious/helpers/handlers.cr:158:7 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from src/invidious/helpers/static_file_handler.cr:189:11 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'\n  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'\n  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'process'\n  from /usr/share/crystal/src/http/server.cr:498:5 in '->'\n  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'\n  from ???\n"}`
**Screenshots**

**Additional context**
this might be due to:
1.) rate limit
2.) a youtube change

**Comment by unixfox** [comment]
Which instance did you use? Or are you running invidious locally?

**Comment by D3V1LC0D3R** [comment]
@unixfox im running a little instance on a vps behind a nginx proxy

**Comment by SamantazFox** [comment]
@D3V1LC0D3R What version are you using (this can be found in the footer of any page generated by invidious)? Can you also provide a the ID of the video where this hapenned? thanks :)

**Comment by D3V1LC0D3R** [comment]
@SamantazFox 
version:  Current version: 2021.07.16-095c894 @ (HEAD detached from 0.20.1)

Example path: ```/api/v1/comments/whLho7HfN4Q?format=html&hl=en-US&thin_mode=false&continuation=EkMSC3doTGhvN0hmTjRRyAEA4AEBogINKP___________wFAAMICHQgEGhdodHRwczovL3d3dy55b3V0dWJlLmNvbSIAGAYyTRpLEhpVZ3pLbnBoWmhMaWNUYko0aG5ONEFhQUJBZyICCAAqGFVDWkhwSUZNZm9KSl8xUXhOR0xKVHp5QTILd2hMaG83SGZONFFAAUgK&action=action_get_comment_replies```

**Comment by SamantazFox** [comment]
Extracted/formatted the interesting part of the error message. Looks like a youtube change:
```
Missing hash key: "url" (KeyError)
 from /usr/share/crystal/src/hash.cr:1030:9 in '[]'
 from src/invidious/comments.cr:576:71 in 'content_to_comment_html'
 from src/invidious/comments.cr:539:43 in 'parse_content'
 from src/invidious/comments.cr:149:69 in 'fetch_youtube_comments'
 from src/invidious.cr:2079:18 in '->'
 from lib/kemal/src/kemal/route.cr:13:9 in '->'
```

**Comment by unixfox** [comment]
> Extracted/formatted the interesting part of the error message. Looks like a youtube change:
> ```
> Missing hash key: "url" (KeyError)
>  from /usr/share/crystal/src/hash.cr:1030:9 in '[]'
>  from src/invidious/comments.cr:576:71 in 'content_to_comment_html'
>  from src/invidious/comments.cr:539:43 in 'parse_content'
>  from src/invidious/comments.cr:149:69 in 'fetch_youtube_comments'
>  from src/invidious.cr:2079:18 in '->'
>  from lib/kemal/src/kemal/route.cr:13:9 in '->'
> ```

That's why #2217 is here but there still some remaining bugs to find.

**Comment by SamantazFox** [comment]
bump @D3V1LC0D3R! Does it still happen? I can't reproduce here.

**Comment by unixfox** [comment]
Closing as it is most likely it got fixed by #2217, please reopen if you still encounter the issue.
